### PR TITLE
show Job name when checking infrabox.json format failed

### PR DESCRIPTION
**Pull request by BigNuoLi**
While using the "infrabox -f infrabox.json validate" to check the infrabox project json file format, when there is invalid property name found, this tool does not show which Job contains the property name.
The fixing here will show the job name which has the invalid property name.

### CODE REVIEW

File: pyinfrabox/utils.py [hunk 1]
```diff
@@ -22,21 +22,29 @@ def check_text(t, path, allowEmpty=False):
...
+
 def check_allowed_properties(d, path, allowed):
     if not isinstance(d, dict):
         raise ValidationError(path, "must be an object")
 
     for key in d:
         if key not in allowed:
-            raise ValidationError(path, "invalid property '%s'" % key)
+            if 'name' in d.keys():
+                raise ValidationError(path + '(' + d['name'] + ')', "invalid property '%s'" % key)
```
Comment (meranos) [inline pyinfrabox/utils.py:33]: For readability
```python
'%s(%s)' % (path, d['name])
```
(same on line 45)

Comment (BigNuoLi) [reply pyinfrabox/utils.py]: Added the check for pyyaml version;
Re-format the message output;

File: infraboxcli/validate.py [hunk 2]
```diff
@@ -19,7 +19,10 @@ def validate_infrabox_file(args):
             data = json.load(f)
         except ValueError:
             f.seek(0)
-            data = yaml.load(f)
+            if sys.version_info.major == 2:
```
[the remark this answers is not in the document]
Comment (BigNuoLi) [reply infraboxcli/validate.py]: add checking yaml version code

**Comment by BigNuoLi** [comment]
> Thanks for suggesting this change. Having that additional information in the error message is definitely helpful 👍

Thanks.

**Review by meranos — APPROVED** [review]

**Pull request merged**